src/middlewared/middlewared/plugins/iscsi_/extents.py

```diff
@@ -411,9 +411,20 @@ def save(self, data, schema_name, verrors):
                     f'Failed to create {dirs} with error: {e}'
                 )
 
-            # create the extent
+            # create the extent, or perhaps extend it
             if not os.path.exists(path):
+                # create the extent
                 subprocess.run(['truncate', '-s', str(data['filesize']), path])
+            else:
+                if old:
+                    old_size = int(old['filesize'])
+                    new_size = int(data['filesize'])
+                    # For now only allow expansion, we might remove this
+                    # restriction later if the webui adds a check.
```

Comment:
We should raise a `ValidationError` to prevent foot-shooting in the event uses the webUI to shrink an extent. There are too many "gotchas" there and too many edge-case scenarios where that can get out of hand.

Reply:
Thanks.  Done.